base/point.h

```diff
@@ -25,13 +25,19 @@ class Point {
    * \brief Standard destructor.
    */
   virtual ~Point();
-  /** General accessor method for the underlying data.
+
+  /** General accessor method for each feature of the underlying data.
    * \param feature_id The coordinate of interest. Indexing is implementation
    * specific.
    * \return The value of the coordinate specified in the input.
    */
   virtual double GetFeatureValue(unsigned feature_id) const = 0;
 
+  /** General accessor method for the label of the underlying data.
+   * \return The label of the object.
+   */
+  virtual double GetLabel() const = 0;
```

Comment:
What does "Label" represent? Is it the class, if we think about a classification problem? If so, maybe this function should only exists in a future "Document" class, as we may try to work with some datasets with no label (Unsupervised learning) in the future.
If Label is some kind of id of given point, then its ok. I think it's good to have something to identify it, besides only its future position in a vector of a Collection